src/core/memory.c

```diff
@@ -104,3 +123,48 @@ void *allocate_huge_page(int size)
   return NULL;
 }
 
+// SIGSEGV handler: attempts map packet memory on demand
+void allocate_on_sigsegv(int sig, siginfo_t *si, void *unused)
+{
+  uint64_t address = (uint64_t)si->si_addr;
+  if ((address & 0x500000000000ULL) != 0x500000000000ULL) {
+    // This is not DMA memory: die.
+    //exit(139);
```

Comment:
I believe that the correct way to exit here would be to re-signal the SIGSEGV.  Something like
```
signal(SIGSEGV, SIG_DFL);
return kill(getpid(), SIGSEGV);
```

See http://www.cons.org/cracauer/sigint.html for a detailed explanation.